ios/FluentUI/Vnext/Avatar/AvatarGroup.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+import SwiftUI
+
+@objc public enum MSFAvatarGroupStyle: Int, CaseIterable {
+    case stack
+    case pile
+}
+
+/// Properties that make up AvatarGroup content
+@objc public class MSFAvatarGroupState: NSObject, ObservableObject {
+    @objc @Published public var avatars: [MSFAvatarState] = []
+    @objc @Published public var maxDisplayedAvatars: UInt32 = UInt32.max
+    @objc @Published public var overflowCount: UInt = 0
+}
+
+/// View that represents the AvatarGroup
+public struct AvatarGroup: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var state: MSFAvatarGroupState
+    @ObservedObject var tokens: MSFAvatarGroupTokens
+
+    init(style: MSFAvatarGroupStyle, size: MSFAvatarSize) {
+        self.state = MSFAvatarGroupState()
+        self.tokens = MSFAvatarGroupTokens(style: style, size: size)
+    }
+
+    public var body: some View {
+        let size = tokens.size.size
+        let x: CGFloat = size + tokens.interspace - tokens.ringOuterGap - tokens.ringThickness
+        let avatars = state.avatars
+        let maxDisplayedAvatars = avatars.prefix(Int(state.maxDisplayedAvatars)).count
+        let overflowCount = (avatars.count > maxDisplayedAvatars ? UInt(avatars.count - maxDisplayedAvatars) : 0) + state.overflowCount
+        HStack(spacing: 0) {
+            ForEach(0 ..< maxDisplayedAvatars, id: \.self) { index in
+                let modify = tokens.style == .stack && (overflowCount > 0 || index + 1 < maxDisplayedAvatars)
```

Comment:
Can we have a more descriptive name for this variable? 
Something like "shouldApplyCutout" or "needsCutout"?